File: stack.py
class Stack:
	def __init__(self, size):
		self.size = size
		self.top = -1
		self.data = []

	def is_full(self):
		return len(self.data) == self.size

	def is_empty(self):
		return self.top == -1

	def increment_top(self):
		self.top += 1

	def decrement_top(self):
		self.top -= 1

	def push(self, x):
		if self.is_full():
			print "Stack is full"
		else:
			self.increment_top()
			self.data.append(x)
			return self.data

	def pop(self):
		if self.is_empty():
			print "Stack is empty"
		else:
			self.decrement_top()
			return self.data.pop()

File: stack.rb
class Stack
  attr_accessor :data

  def intiialize(size = 10)
    @size = size
  end

  def data
    @data ||= []
  end

  def empty?
    top == -1
  end

  def full?
    data.length == @size
  end

  def add_top
    @top ||= -1
    @top += 1
  end

  def less_top
    @top ||= -1
    @top -= 1
  end

  def push(x)
    unless full?
      add_top
      data.push(x)
    else
      puts "Stack is Full"
    end
  end

  def pop
    unless empty?
      less_top
      data.pop
    else
      puts "Stack is empty"
    end
  end

end
